tests/recognizer-tests.js

```diff
@@ -129,6 +129,16 @@ test("A dynamic route recognizes", function() {
   equal(router.recognize("/zoo/baz"), null);
 });
 
+test("A dynamic route with encoded params", function() {
+  var handler = {};
+  var router = new RouteRecognizer();
+  router.add([{ path: "/foo/:bar", handler: handler }]);
+
+  equal(router.recognize("/foo/bar%20baz")[0].params.bar, "bar baz");
+  equal(router.recognize("/foo/bar%23baz")[0].params.bar, "bar#baz");
+  equal(router.recognize("/foo/bar%3fbaz")[0].params.bar, "bar?baz");
+});
+
```

Comment:
Well probably the `(` and `)` denoting the optional part would never make it into the `add` function. You can imagine a naive implementation that might simply iterate over all possible combinations and add each individual one. In this case we'd add `/lol` and `/lol/(wat)` (which have already been decoded).